### PR TITLE
Switch to  BleChessProtocol

### PR DESCRIPTION
**Pull request by vovagorodok**

### CODE REVIEW

**Comment by dkwach** [comment]
We need update README:
src/externalDevice/README.md

File: src/externalDevice/BleChessProtocol.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+  }
+}
+
+class Init extends BleChessState {
+  onEnter() {
+    this.transitionTo(new CheckFeatureMsg)
+  }
+}
+
+class CheckFeatureMsg extends BleChessState {
```
Comment (dkwach) [inline src/externalDevice/BleChessProtocol.ts:48]: Can we merge it with Init? Do we need to two states: Init and CheckFeatureMsg

**Comment by vovagorodok** [comment]
> We need update README: src/externalDevice/README.md

Documentation will be changed in future. Let's finish functional part

File: src/externalDevice/BleChessProtocol.ts (hunk 1 above)
Comment (dkwach) [inline src/externalDevice/BleChessProtocol.ts:48]: Maybe just `CheckFeature` or `InitFeature` instead `CheckFeatureMsg`

File: src/externalDevice/BleChessProtocol.ts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    else super.onPeripheralCommand(cmd)
+  }
+}
+
+class CheckFeatureLastMove extends BleChessState {
+  onEnter() {
+    sendCommandToPeripheral('feature last_move')
+  }
+  onPeripheralCommand(cmd: string) {
+    const isCreated = isCentralStateCreated(this.getState())
```
Comment (dkwach) [inline src/externalDevice/BleChessProtocol.ts]: can you explain isCreated?

**Comment by dkwach** [comment]
> > We need update README: src/externalDevice/README.md
> 
> Documentation will be changed in future. Let's finish functional part

Shall we remove readme for now? or remove all uci cecp elements from it?

File: src/externalDevice/BleChessProtocol.ts (hunk 2 above)
Comment (vovagorodok) [reply src/externalDevice/BleChessProtocol.ts]: it means that game whas started bifore we connect to bluetooth peripheral

Usecase:
```
- start game
- go to settings and connect via bluetooth
- continue playing game
```

without it, we mast restart game

File: src/externalDevice/BleChessProtocol.ts (hunk 1 above)
Comment (vovagorodok) [reply src/externalDevice/BleChessProtocol.ts:48]: But this state for chacking `msg` feauture

Comment (vovagorodok) [reply src/externalDevice/BleChessProtocol.ts:48]: Yes, because we send next feature checking after `ok` sequentially

**Comment by vovagorodok** [comment]
> > > We need update README: src/externalDevice/README.md
> > 
> > 
> > Documentation will be changed in future. Let's finish functional part
> 
> Shall we remove readme for now? or remove all uci cecp elements from it?

Lets release app with functional part. I'll update documentation  in next PR

File: src/externalDevice/BleChessProtocol.ts (hunk 2 above)
Comment (dkwach) [reply src/externalDevice/BleChessProtocol.ts]: Maybe let's rename it to isGameStartedBefore

Comment (vovagorodok) [reply src/externalDevice/BleChessProtocol.ts]: `isCentralStateCreated()` has better naming consistency with  `onCentralStateCreated()`  and `onCentralStateChanged()`

**Review by dkwach — APPROVED** [review]

**Pull request merged**